calyx/src/analysis/order_analysis.rs

```diff
@@ -0,0 +1,164 @@
+use crate::analysis::ReadWriteSet;
+use crate::ir;
+
+#[derive(Default)]
+///Primarily used to help determine the order cells are executed within
+///the group
+pub struct OrderAnalysis {
```

Comment:
Is this implementing the same logic as `InferStaticTiming`? If so, we should generalize those two

Reply:
There is some overlap between,  `mem_wrt_dep_graph` of `InferStaticTiming` and `update_map` of `OrderAnalysis` but there are differences as well and in my opinion it's not worth it to try to generalize.